### DISCUSSION TITLE
Split endpoints in multiple files with working auth?

### DISCUSSION THREAD

**Discussion by hsul-git** [fastapi-users/fastapi-users#1091] [Q&A]
Hey everyone,

i seperated my endpoints in a other file and used includ_router() to get them from there into the apps.py. The endpoints working just fine, but the authentication seems to be broken now. Do i have to pass down the credentials? In the response i got no bearer token back and got the error "Auth ErrorError: Not Found".

I did it like in this tutorial: https://levelup.gitconnected.com/5-advance-features-of-fastapi-you-should-try-7c0ac7eebb3e (3. Separate your FastAPI Routes into different files)

Here is my current code:

app.py
```
from app.routes.administration import admin
from app.routes.data import dat

app = FastAPI()
app.include_router(admin)
app.include_router(dat)


@app.on_event("startup")
async def on_startup():
await create_db_and_tables()
```

administration.py
```
admin = APIRouter()


admin.include_router(
    fastapi_users.get_auth_router(auth_backend), tags=["auth"],
)
```

When i define it like this (single file) in the app.py, the authentication works:

app.py
```
current_active_user = fastapi_users.current_user(active=True)

app = FastAPI()

app.include_router(
    fastapi_users.get_auth_router(auth_backend), prefix="/auth/jwt", tags=["auth"]
)

@app.on_event("startup")
async def on_startup():
    # Not needed if you setup a migration system like Alembic
    await create_db_and_tables()
```

Do you guys have a hint for me? Whats wrong?

Thank you.

**Comment by frankie567** [comment]
Don't know if it's a mistake in your copy/paste but in your `administration.py` file, you named your router `adm`:

```py
adm = APIRouter()
```

But in `app.py` you import `admin`:

```py
from app.routes.administration import admin
```

It should probably be:

```py
from app.routes.administration import adm
```

**Comment by hsul-git** [comment]
Was a mistake by c&p. 

I initialy missed the "prefix="/auth/jwt"".  Deleted it and didnt knew that the prefix is essential, because it is defined in the deeper code. 

But thank you for your support.
